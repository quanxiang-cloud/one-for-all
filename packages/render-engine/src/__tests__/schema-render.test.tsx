// import React, { useEffect, useRef, useState } from 'react';
// import { render } from '@testing-library/react';
// import { APISpecAdapter } from '@one-for-all/api-spec-adapter';
// import type * as SchemaSpec from '@one-for-all/schema-spec';

// import { RenderEngineCTX } from '../types';
// import SchemaRender from '../schema-render';

// const schema: SchemaSpec.Schema = {
//   apiStateSpec: {},
//   sharedStatesSpec: {},
//   node: {
//     id: 'id',
//     type: 'html-element',
//     name: 'div',
//   },
// };

// const dummyAdapter: APISpecAdapter = {
//   build: () => ({ url: '', method: '' }),
// };

// function Dummy(): JSX.Element {
//   const ref = useRef<RenderEngineCTX>(null);
//   const [state, setState] = useState<RenderEngineCTX | undefined>();

//   useEffect(() => {
//     if (!ref.current) {
//       return;
//     }

//     setState(ref.current);
//   }, []);

//   return (
//     <>
//       <SchemaRender ref={ref} schema={schema} plugins={{ apiSpecAdapter: dummyAdapter }} />
//       {state && <div jest-id="state" />}
//     </>
//   );
// }

// test('SchemaRender_should_return_expected_ref', () => {
//   const { container } = render(<Dummy />);
//   expect(container.querySelector('[jest-id="state"]')).toBeTruthy();
// });

test('todo', () => {
  expect(true).toBeTruthy();
});
