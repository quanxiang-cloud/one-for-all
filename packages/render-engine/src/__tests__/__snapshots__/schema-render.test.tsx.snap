// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaRender_should_render_null 1`] = `<div />`;
